strategy/_utils.ts

```diff
@@ -0,0 +1,26 @@
+// Copyright 2018-2020 Gamebridge.ai authors. All rights reserved. MIT license.
+import { Serializable } from "../serializable.ts";
+
+type NewSerializable<T> = T & (new () => Serializable);
+type FunctionSerializable = () => Serializable;
+/** for strategies */
+export type StrategyTypeArgument<T> = NewSerializable<T> | FunctionSerializable;
+
+/** for strategy values */
+export function isObject(obj: any): obj is Record<string, any> {
+  return Object.prototype.toString.call(obj) === "[object Object]";
+}
+
+/** for strategy type arguments */
+function isNewable<T>(
+  type: NewSerializable<T> | FunctionSerializable,
+): type is NewSerializable<T> {
+  return !!type.name;
```

Comment:
Neat.

I did some Googling, what would you think about https://stackoverflow.com/a/57862312/810525
```
function isConstructor(func) {
    return (func && typeof func === "function" && func.prototype && func.prototype.constructor) === func;
}
```
The relevant bit of that is probably `func.prototype && (func.prototype.constructor === func)`.